e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml

```diff
@@ -0,0 +1,221 @@
+         get_url:
+           url: "{{ sql_test_link }}"
+           dest: "{{ result_kube_home.stdout }}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: 8a) Create the configmap test
+         command: kubectl create configmap sqltest --from-file=sql-test.sh -n {{ namespace }}
+         args:
+           executable: /bin/bash
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: 9) Obtaining the percona liveliness application yaml
+         get_url:
+           url: "{{ percona_link }}"
+           dest: "{{ result_kube_home.stdout }}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: 9a) Create percona deployment with OpenEBS storage
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_task.yml"
+         vars:
+           app_yml: "{{ percona_files.0 }}"
+           ns: {{ namespace }}
+
+       - name: 9b) Check whether percona application is running
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+           app: percona
+           ns: {{ namespace }}
+
+       - name: 9c) Check Replica pod status
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+           app: pvc
+           ns: {{ namespace }}
+
+       - name: 10) Failing the additional disk
+         shell: echo 1 > /sys/block/{{ result_disk.stdout }}/device/delete
+         args:
+           executable: /bin/bash
+         become: true
+         delegate_to: "{{ groups['kubernetes-kubeminions'].0 }}"
+
+       - name: 11) List the available scsi host
+         command: ls /sys/class/scsi_host/
+         args:
+           executable: /bin/bash
+         become: true
+         register: result_host
+         delegate_to: "{{ groups['kubernetes-kubeminions'].0 }}"
+
+       - name: 11a) Rescan the disk
+         shell: echo "- - -" > /sys/class/scsi_host/{{ item }}/scan
+         args:
+           executable: /bin/bash
+         become: true
+         with_items: "{{ result_host.stdout_lines }}"
+         delegate_to: "{{ groups['kubernetes-kubeminions'].0 }}"
+
+       - name: 11b) Get the disk name using UUID
+         shell: blkid | grep {{ uuid }} | awk '{print $1}' | cut -d "/" -f3 | cut -d ":" -f1
+         args:
+           executable: /bin/bash
+         become: true
+         register: result_scan_disk
+         delegate_to: "{{ groups['kubernetes-kubeminions'].0 }}"
+
+       - name: 12) Remount the disk
+         shell: mount /dev/{{ result_scan_disk.stdout }} {{ result_mountpoint.stdout }}
+         args:
+           executable: /bin/bash
+         become: true
+         delegate_to: "{{ groups['kubernetes-kubeminions'].0 }}"
+         tags:
+         - skip_ansible_lint
+
+       - name: 13) Check Replica pod status
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+           app: pvc
+           ns: {{ namespace }}
```

Comment:
Once the replica pod is running, there has to be a check to see that it has been added back in RW mode, i.e., if dirty data sync has occurred.